libs/crypto/include/crypto/ecdsa_signature.hpp

```diff
@@ -165,40 +165,41 @@ class ECDSASignature
   };
 
   //* For safe (noexcept) moving semantic constuctor
-  ECDSASignature(byte_array::ConstByteArray &&hash, shrd_ptr_type<ECDSA_SIG> &&signature_ECDSA_SIG,
-                 byte_array::ConstByteArray &&signature)
+  ECDSASignature(byte_array::ConstByteArray &&  hash,
+                 SharedPointerType<ECDSA_SIG> &&SignatureECDSAPtr,
+                 byte_array::ConstByteArray &&  signature)
     : hash_{std::move(hash)}
-    , signature_ECDSA_SIG_{std::move(signature_ECDSA_SIG)}
+    , SignatureECDSAPtr_{std::move(SignatureECDSAPtr)}
     , signature_{std::move(signature)}
   {}
 
   template <eECDSAEncoding BIN_FORMAT>
-  static ECDSASignature safeMoveConstruct(ecdsa_signature_type<BIN_FORMAT> &&from)
+  static ECDSASignature safeMoveConstruct(EcdsaSignatureType<BIN_FORMAT> &&from)
   {
     byte_array::ConstByteArray signature{
-        Convert(from.signature_ECDSA_SIG_, signatureBinaryDataFormat)};
+        Convert(from.SignatureECDSAPtr_, signatureBinaryDataFormat)};
 
-    return ECDSASignature{std::move(from.hash_), std::move(from.signature_ECDSA_SIG_),
+    return ECDSASignature{std::move(from.hash_), std::move(from.SignatureECDSAPtr_),
                           std::move(signature)};
   }
 
   template <eECDSAEncoding BIN_ENC, point_conversion_form_t POINT_CONV_FORM>
-  ECDSASignature(private_key_type<BIN_ENC, POINT_CONV_FORM> const &private_key,
-                 byte_array::ConstByteArray const &                data_to_sign,
-                 const eBinaryDataType data_type = eBinaryDataType::data)
-    : hash_{data_type == eBinaryDataType::data ? Hash<hasher_type>(data_to_sign) : data_to_sign}
-    , signature_ECDSA_SIG_{CreateSignature(private_key, hash_)}
-    , signature_{Convert(signature_ECDSA_SIG_, signatureBinaryDataFormat)}
+  ECDSASignature(PrivateKeyType<BIN_ENC, POINT_CONV_FORM> const &private_key,
+                 byte_array::ConstByteArray const &              data_to_sign,
+                 const eBinaryDataType                           DataType = eBinaryDataType::data)
+    : hash_{DataType == eBinaryDataType::data ? Hash<HasherType>(data_to_sign) : data_to_sign}
+    , SignatureECDSAPtr_{CreateSignature(private_key, hash_)}
+    , signature_{Convert(SignatureECDSAPtr_, signatureBinaryDataFormat)}
```

Comment:
This private ctor is better split in two overloads IMO, one for hash other for data, and used accordingly. This would simplify `hash_` initialization. An empty tag struct could be used to differentiate between the two, rather than enum.

Reply:
I would propose to simplify even further by just moving the ` Hash<HasherType>` function to `Sign` and just have a single constructor.

Reply:
(line 114 - 126)